engine/src/api/workflow/agents/base.py

```
@@ -241,5 +241,41 @@ async def _get_llm_response(
+            raw = await self.llm_client.structured_chat_completion(
+                messages=messages, schema=schema, temperature=temperature
+            )
+
+            # If the caller passed a Pydantic model, validate here
+            if isinstance(schema, type) and issubclass(schema, BaseModel):
+                try:
+                    # Validate and convert to dict
+                    return schema.model_validate(raw).model_dump()
+                except Exception as e:
+                    logger.error(f"Error validating LLM response against schema: {str(e)}")
+                    raise ValueError(f"LLM response failed schema validation: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)